ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0104__rhods_operator/0110__service_mesh.robot

```diff
@@ -58,15 +61,119 @@ Validate Service Mesh State Removed
     [Teardown]    Set Service Mesh State To Managed And Wait For CR Ready
     ...           ${SERVICE_MESH_CR_NAME}    ${SERVICE_MESH_CR_NS}    ${OPERATOR_NS}
 
+Validate Service Mesh Control Plane Already Created
+    [Documentation]    This Test Case validates that only one ServiceMeshControlPlane is allowed to be installed per project/namespace
+    [Tags]      Operator        Tier3       RHOAIENG-2517
+    Fetch Image Url And Update Channel
+    Check Whether DSC Exists
+    Fetch Cluster Type By Domain
+    IF    "${CLUSTER_TYPE}" == "selfmanaged"
+        Uninstall RHODS In Self Managed Cluster
+        Create Smcp From Template
+        Install RHODS In Self Managed Cluster Using CLI     ${CLUSTER_TYPE}     ${IMAGE_URL}
+    ELSE IF    "${CLUSTER_TYPE}" == "managed"
+        Uninstall RHODS In OSD
+        Create Smcp From Template
+        Install RHODS In Managed Cluster Using CLI      ${CLUSTER_TYPE}     ${IMAGE_URL}
+    END
+    Operator Deployment Should Be Ready
+    # Go check the Operator logs for the error message: denied the request: only one service mesh may be installed per project/namespace
+    Verify Pod Logs Contain Error
+    [Teardown]      Teardown Service Mesh Control Plane Already Created
+
 
 *** Keywords ***
 Suite Setup
     [Documentation]    Suite Setup
     RHOSi Setup
     Wait Until Operator Ready    ${SERVICE_MESH_OPERATOR_DEPLOYMENT_NAME}    ${SERVICE_MESH_OPERATOR_NS}
-    Wait Until Operator Ready    ${OPERATOR_DEPLOYMENT_NAME}    ${OPERATOR_NS}
-    Wait For DSCI Ready State    ${DSCI_NAME}    ${OPERATOR_NS}
+    Wait Until Operator Ready    ${OPERATOR_DEPLOYMENT_NAME}    ${OPERATOR_NAMESPACE}
+    Wait For DSCI Ready State    ${DSCI_NAME}    ${OPERATOR_NAMESPACE}
 
 Suite Teardown
     [Documentation]    Suite Teardown
     RHOSi Teardown
+
+Teardown Service Mesh Control Plane Already Created
+    # Cleanup the SMCP
+    Delete Smcp
+    # Cleanup Olminstall dir
+    Cleanup Olm Install Dir
+    IF      ${DSC_EXISTS} == True
+        Apply DataScienceCluster CustomResource     ${DSC_NAME}
```

Comment:
I don't understand, if DSCI exists, why applying it again?

Reply:
It will rebuild DSC if previously existed, in order not to interrupt other Test Cases execution